platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsToolingApiBuildActionIntegrationTest.groovy

```diff
@@ -621,4 +621,17 @@ class IsolatedProjectsToolingApiBuildActionIntegrationTest extends AbstractIsola
         model4.message == "It works from project :a"
     }
 
+    def "does not cache execution of BuildAction when it fails"() {
+        when:
+        withIsolatedProjects()
+        runBuildActionFails(new FailingBuildAction())
+
+        then:
+        fixture.assertNoConfigurationCache()
+        failureDescriptionContains("Build action expectedly failed")
+
+        // TODO:isolated should not contain this output https://github.com/gradle/gradle/issues/27476
+        failure.assertOutputContains("Configuration cache entry stored.")
```

Comment:
🤔 Do we really need this assertion after the `fixture.assertNoConfigurationCache()` above?

Reply:
I don't know. I think that depends on how we resolve https://github.com/gradle/gradle/issues/27476, which will alter the behavior, and then we can decide whether to remove the assertion here, or that we are printing something else in this case.

I would keep it for now. I don't think it's take a lot of time to fix the abovementioned issue.